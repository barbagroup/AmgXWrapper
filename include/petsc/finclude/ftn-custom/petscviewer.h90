#if !defined(PETSC_USE_FORTRAN_MODULES)
#include "petsc/finclude/ftn-custom/petscviewerdef.h90"
#endif
#include "petsc/finclude/ftn-custom/petscviewerhide.h90"

      Interface PetscViewerBinaryWrite
        Subroutine PetscViewerBinaryWriteInt(v,a,cnt,tmp,ierr)
          USE_PETSC_HIDE
          PETSCVIEWER_HIDE v
          PetscInt a(*)
          PetscInt cnt
          PetscBool  tmp
          PetscErrorCode ierr
        End Subroutine

        Subroutine PetscViewerBinaryWriteScalar(v,a,cnt,tmp,ierr)
          USE_PETSC_HIDE
          PETSCVIEWER_HIDE v
          PetscScalar a(*)
          PetscInt cnt
          PetscBool  tmp
          PetscErrorCode ierr
        End Subroutine

#if defined(PETSC_USE_COMPLEX)
        Subroutine PetscViewerBinaryWriteReal(v,a,cnt,tmp,ierr)
          USE_PETSC_HIDE
          PETSCVIEWER_HIDE v
          PetscReal a(*)
          PetscInt cnt
          PetscBool  tmp
          PetscErrorCode ierr
        End Subroutine
#endif

        Subroutine PetscViewerBinaryReadInt(v,a,cnt,ierr)
          USE_PETSC_HIDE
          PETSCVIEWER_HIDE v
          PetscInt a(*)
          PetscInt cnt
          PetscErrorCode ierr
        End Subroutine

        Subroutine PetscViewerBinaryReadScalar(v,a,cnt,ierr)
          USE_PETSC_HIDE
          PETSCVIEWER_HIDE v
          PetscScalar a(*)
          PetscInt cnt
          PetscErrorCode ierr
        End Subroutine

#if defined(PETSC_USE_COMPLEX)
        Subroutine PetscViewerBinaryReadReal(v,a,cnt,ierr)
          USE_PETSC_HIDE
          PETSCVIEWER_HIDE v
          PetscReal a(*)
          PetscInt cnt
          PetscErrorCode ierr
        End Subroutine
#endif

      End Interface

