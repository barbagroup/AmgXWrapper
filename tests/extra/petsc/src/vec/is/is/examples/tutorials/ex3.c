
static char help[] = "Demonstrates creating a blocked index set.\n\n";

/*T
    Concepts: index sets^creating a block index set;
    Concepts: IS^creating a block index set;

    Description:  Creates an index set based on blocks of integers. Views that index set
    and then destroys it.
T*/

#include <petscis.h>
#include <petscviewer.h>

#undef __FUNCT__
#define __FUNCT__ "main"
int main(int argc,char **argv)
{
  PetscErrorCode ierr;
  PetscInt       i,n = 4, inputindices[] = {0,1,3,4},bs = 3,issize;
  const PetscInt *indices;
  IS             set;
  PetscBool      isblock;

  ierr = PetscInitialize(&argc,&argv,(char*)0,help);CHKERRQ(ierr);

  /*
    Create a block index set. The index set has 4 blocks each of size 3.
    The indices are {0,1,2,3,4,5,9,10,11,12,13,14}
    Note each processor is generating its own index set
    (in this case they are all identical)
  */
  ierr = ISCreateBlock(PETSC_COMM_SELF,bs,n,inputindices,PETSC_COPY_VALUES,&set);CHKERRQ(ierr);
  ierr = ISView(set,PETSC_VIEWER_STDOUT_SELF);CHKERRQ(ierr);

  /*
    Extract indices from set.
  */
  ierr = ISGetLocalSize(set,&issize);CHKERRQ(ierr);
  ierr = ISGetIndices(set,&indices);CHKERRQ(ierr);
  ierr = PetscPrintf(PETSC_COMM_SELF,"Printing indices directly\n");CHKERRQ(ierr);
  for (i=0; i<issize; i++) {
    ierr = PetscPrintf(PETSC_COMM_SELF,"%D\n",indices[i]);CHKERRQ(ierr);
  }
  ierr = ISRestoreIndices(set,&indices);CHKERRQ(ierr);

  /*
    Extract the block indices. This returns one index per block.
  */
  ierr = ISBlockGetIndices(set,&indices);CHKERRQ(ierr);
  ierr = PetscPrintf(PETSC_COMM_SELF,"Printing block indices directly\n");CHKERRQ(ierr);
  for (i=0; i<n; i++) {
    ierr = PetscPrintf(PETSC_COMM_SELF,"%D\n",indices[i]);CHKERRQ(ierr);
  }
  ierr = ISBlockRestoreIndices(set,&indices);CHKERRQ(ierr);

  /*
    Check if this is really a block index set
  */
  ierr = PetscObjectTypeCompare((PetscObject)set,ISBLOCK,&isblock);CHKERRQ(ierr);
  if (!isblock) SETERRQ(PETSC_COMM_SELF,1,"Index set is not blocked!");

  /*
    Determine the block size of the index set
  */
  ierr = ISGetBlockSize(set,&bs);CHKERRQ(ierr);
  if (bs != 3) SETERRQ(PETSC_COMM_SELF,1,"Block size is not 3!");

  /*
    Get the number of blocks
  */
  ierr = ISBlockGetLocalSize(set,&n);CHKERRQ(ierr);
  if (n != 4) SETERRQ(PETSC_COMM_SELF,1,"Number of blocks not 4!");

  ierr = ISDestroy(&set);CHKERRQ(ierr);
  ierr = PetscFinalize();
  return 0;
}


