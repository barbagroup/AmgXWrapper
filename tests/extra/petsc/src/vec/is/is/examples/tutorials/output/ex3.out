IS Object: 1 MPI processes
  type: block
Block size 3
Number of block indices in set 4
The first indices of each block are
Block 0 Index 0
Block 1 Index 1
Block 2 Index 3
Block 3 Index 4
Printing indices directly
0
1
2
3
4
5
9
10
11
12
13
14
Printing block indices directly
0
1
3
4
